### PR TITLE
Filter out builds removed from main.

### PR DESCRIPTION
**Pull request by godofredoc**
Filter out builds removed from main when running for a different branch.

Bug: https://github.com/flutter/flutter/issues/122235
Bug: https://github.com/flutter/flutter/issues/122204

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by godofredoc** [comment]
@CaseyHillers thank you for providing the test fixes.

File: app_dart/lib/src/service/scheduler.dart [hunk 1]
```diff
@@ -121,15 +121,22 @@ class Scheduler {
...
+    final Commit totCommit = await generateTotCommit(slug: commit.slug, branch: Config.defaultBranch(commit.slug));
+    final CiYaml totYaml = await getCiYaml(totCommit);
     final CiYaml ciYaml = await getCiYaml(commit);
     final List<Target> initialTargets = ciYaml.getInitialTargets(ciYaml.postsubmitTargets);
-    final List<Task> tasks = targetsToTask(commit, initialTargets).toList();
+    // Filter targets.
+    final List<String> totTargetNames = totYaml.postsubmitTargets.map((Target target) => target.value.name).toList();
+    final List<Target> filteredTargets =
+        initialTargets.where((Target target) => totTargetNames.contains(target.value.name)).toList();
+    final List<Task> tasks = targetsToTask(commit, filteredTargets).toList();
```
Comment (CaseyHillers) [inline app_dart/lib/src/service/scheduler.dart:139]: Should this logic be upstreamed in https://cs.opensource.google/flutter/cocoon/+/main:app_dart/lib/src/model/ci_yaml/ci_yaml.dart;l=72?

Then it only needs to be implemented once for both presubmit and postsubmit.

Comment (godofredoc) [reply app_dart/lib/src/service/scheduler.dart]: Done!

**Review by CaseyHillers — APPROVED** [review]
flutter-lgtm

**Pull request merged**